nengo/spa/basalganglia.py

```diff
@@ -141,3 +143,36 @@ def add_dot_input(self, index, source, symbol, scale):
         with self.spa:
             nengo.Connection(output, self.input[index:index+1],
                              transform=transform, synapse=self.input_synapse)
+
+    def add_scalar_input(self, index, source):
+        """ Add a scalar input that will vary over time.
+
+        This would be used for such a thing as the ouput of the Compare module.
+
+        Parameters
+        ----------
+        index : int
+            the index of the action
+        source : Source
+            the module output to read from
+        symbol : Source
+            the semantic pointer to compute the dot product with
+        scale : float
+            a scaling factor to be applied to the result
+        """
+        output, _ = self.spa.get_module_output(source.name)
+        if(output.size_out != 1):
+            raise NotImplementedError("Only sources with a dimension"
+                                      "of 1 can be scalar inputs")
+
+        try:
+            scale = float(eval(source.transform.symbol))
+        except ValueError:
+            raise NotImplementedError("Transform can only be a scalar value"
+                                      "the value %s is invalid"
+                                      % source.transform.symbol)
+
+        with self.spa:
+            nengo.Connection(output, self.input[index:index+1],
```

Comment:
Gah, never mind...  Hmm, weird, I thought pep8 enforced that spacing, but it doesn't look like it does...  Basically, I didn't notice that the other code was also bad this way.  It looks like it isn't technically required for pep8, but in the rest of the codebase we are putting spaces around operators, so we should probably be consistent here.  My instinct is that it doesn't matter and leave it how it is, but people have yelled at me for my coding style instincts in the past....  ;)